### DISCUSSION TITLE
Ability to modify the menu bar at runtime

### DISCUSSION THREAD

**Discussion by danmatthews** [NativePHP/laravel#142] [Ideas]
I would love to be able to swap the menu bar icon at runtime to show different states (errors/new alerts/updates available etc).

**Comment by bbredewold** [comment]
Yep, that's on my wish list too. Maybe I'll write a PR for this.
Also it would be nice to make the icon optional, for text-only menu-bar-apps :)

**Comment by modernben** [comment]
I think this is a bigger discussion of being able to manipulate the MenuBar in general after boot. If you do this it'll change but currently, there isn't a method to grab the MenuBar instance after boot. Talking with another dev on discord he suggested putting the `$bar` into the service container and then grabbing it later (in a cron job or something) but I am still struggling with how to do that.

```php
$bar = MenuBar::create()
    ->icon(public_path('menuBarIcon.png'));

$bar->icon('something_else.png');
```

**Comment by simonhamp** [comment]
This should now be doable in [nativephp/laravel v0.6.3](https://github.com/NativePHP/laravel/releases/tag/0.6.3) and [nativephp/electron v0.8.6](https://github.com/NativePHP/electron/releases/tag/0.8.6).

Docs incoming.
